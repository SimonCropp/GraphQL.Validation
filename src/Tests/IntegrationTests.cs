using System.Threading.Tasks;
using GraphQL.FluentValidation;
using Xunit;
using Xunit.Abstractions;

public class IntegrationTests :
    XunitApprovalBase
{
    static ValidatorTypeCache typeCache;

    static IntegrationTests()
    {
        typeCache = new ValidatorTypeCache();
        typeCache.AddValidatorsFromAssemblyContaining<IntegrationTests>();
    }

    [Fact]
    public async Task AsyncValid()
    {
        var queryString = @"
{
  asyncQuery
    (
      input: {
        content: ""TheContent""
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task AsyncInvalid()
    {
        var queryString = @"
{
  asyncQuery
    (
      input: {
        content: """"
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task Valid()
    {
        var queryString = @"
{
  inputQuery
    (
      input: {
        content: ""TheContent""
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task Invalid()
    {
        var queryString = @"
{
  inputQuery
    (
      input: {
        content: """"
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task ComplexValid()
    {
        var queryString = @"
{
  complexInputQuery
    (
      input: {
        inner: {
          content: ""TheContent""
        }
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task ComplexInvalid()
    {
        var queryString = @"
{
  complexInputQuery
    (
      input: {
        inner: {
          content: """"
        }
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task ComplexInvalid2()
    {
        var queryString = @"
{
  complexInputQuery
    (
      input: {
        inner: null
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task AsyncComplexValid()
    {
        var queryString = @"
{
  asyncComplexInputQuery
    (
      input: {
        inner: {
          content: ""TheContent""
        }
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }

    [Fact]
    public async Task AsyncComplexInvalid()
    {
        var queryString = @"
{
  asyncComplexInputQuery
    (
      input: {
        inner: {
          content: """"
        }
      }
    )
  {
    data
  }
}";
        var result = await QueryExecutor.ExecuteQuery(queryString, null, typeCache);
        ObjectApprover.Verify(result);
    }


    public IntegrationTests(ITestOutputHelper output) :
        base(output)
    {
    }
}