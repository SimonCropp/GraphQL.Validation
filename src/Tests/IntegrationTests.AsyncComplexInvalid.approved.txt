{
  data: {
    asyncComplexInputQuery: null
  },
  errors: [
    {
      message: 'Inner: Inner async test failed msg.'
    },
    {
      message: 'Inner.Content: \'Content\' must not be empty.'
    }
  ]
}